x/ibc/client/query.go

```diff
@@ -12,16 +12,28 @@ import (
 	host "github.com/cosmos/cosmos-sdk/x/ibc/24-host"
 )
 
-// QueryTendermintProof performs an ABCI query with the given key and returns the value
-// of the query, the proto encoded merkle proof for the query and the height
-// at which the proof will succeed on a tendermint verifier (one above the
-// returned IAVL version height).
+// QueryTendermintProof performs an ABCI query with the given key and returns
+// the value of the query, the proto encoded merkle proof, and the height at
+// which the proof will succeed on a tendermint verifier (one above the
+// returned IAVL version height). The desired tendermint height to perform
+// the query should be set in the client context. The query will be performed
+// at one below this height (at the IAVL version) in order to obtain the
+// correct merkle proof.
 // Issue: https://github.com/cosmos/cosmos-sdk/issues/6567
 func QueryTendermintProof(clientCtx client.Context, key []byte) ([]byte, []byte, clienttypes.Height, error) {
+	height := clientCtx.Height
+
+	// Use the IAVL height if a valid tendermint height is passed in.
+	// ABCI queries at zero or negative height may have other side affects.
+	if clientCtx.Height > 0 {
```

Comment:
I think this is incorrect. Passing in height `1` would fail.

Reply:
How do height `1` queries work, `1` is the first block height IIRC, so is the state root calculated from genesis state?

Reply:
I don't know the answer to these questions. I can bump this check to `height > 1` but then a clientCtx Height of 1 and 2 would both return the same result which is counter-intuitive.

Reply:
did some testing of this through the testing package. It seems a height of 0 or 1 results in `cannot query with proof when height <= 1` . Note this is from calling `App.Query` directly. It is unclear to me if the same will occur if I query through the ClientCtx. So I guess the check should be `height >  2`? Again this would be slightly strange since height == 2 and height == 3 would return the same result

Reply:
updated to return an error for heights <= `2`